### PR TITLE
Ria 1485 record application

### PR DESCRIPTION
**Pull request by makowalski**
### JIRA link (if applicable) ###
[RIA-1485](https://tools.hmcts.net/jira/browse/RIA-1485)

### Change description ###
record application

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: e2e/features/record-application.feature [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+    And I should see `The legal representative` in the `Application from` field
+    And I should see `Adjourn` in the `Type of application` field
+    And I should see `some application reason` in the `Reason for application` field
+    And I should see `{$TODAY-3|D MMM YYYY}` in the `Date application was made` field
+    And I should see `Granted` in the `Application decision` field
+    And I should see `some application decision reason` in the `Reasons for decision` field
+
+    When I click the `Record application` button
+    Then I should see the text `Unable to proceed because there are one or more callback Errors or Warnings`
+    And I should see an error in summary saying `You can't record application with 'Adjourn' type when case is in 'APPEAL_SUBMITTED' state`
```
Comment (Srihari-Nakkala) [inline e2e/features/record-application.feature]: May be we can cover the "Expedite" scenario

Comment (makowalski) [reply e2e/features/record-application.feature]: I added table for all three application types, where error should be visible

Comment (Srihari-Nakkala) [reply e2e/features/record-application.feature]: 👍

File: e2e/features/case-progression.feature [hunk 2]
```diff
@@ -457,10 +471,24 @@ Feature: Case progression
...
     And I should see the option `Create case summary` for the `Next step` field
     And I should see the option `Add case note` for the `Next step` field
+    And I should see the option `Edit case listing` for the `Next step` field
+    And I should see the option `Record an application` for the `Next step` field
+
+    And I should see the case details
+    And I should see the hearing details
+
+    And I should see the case details
+    And I should see the hearing details
```
Comment (bradsorour) [inline e2e/features/case-progression.feature]: Duplication of the two lines above?

**Pull request merged**